packages/filebrowser-extension/src/index.ts

```diff
@@ -691,6 +693,74 @@ export const launcherToolbarButton: JupyterFrontEndPlugin<void> = {
   }
 };
 
+/**
+ * A plugin to open files from remote URLs
+ */
+const openUrlPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/filebrowser-extension:open-url',
+  autoStart: true,
+  requires: [IFileBrowserFactory, ITranslator],
+  optional: [ICommandPalette],
+  activate: (
+    app: JupyterFrontEnd,
+    factory: IFileBrowserFactory,
+    translator: ITranslator,
+    palette: ICommandPalette | null
+  ) => {
+    const { commands } = app;
+    const trans = translator.load('jupyterlab');
+    const { defaultBrowser: browser } = factory;
+    const command = CommandIDs.openUrl;
+
+    commands.addCommand(command, {
+      label: args =>
+        args.path ? trans.__('Open %1', args.path) : trans.__('Open from URL…'),
```

Comment:
Since the commands uploads the content to the server, we might want to rename the label to something like `Download and Open from URL...`

Reply:
Pinging @jasongrout for his input on the wording.